Editor/AGS.Editor/Components/FontsComponent.cs

```diff
@@ -40,71 +53,63 @@ public override string ComponentID
-                    {
-                        if (item.ID > removingID)
-                        {
-                            if (File.Exists(item.WFNFileName))
-                            {
-                                _agsEditor.RenameFileOnDisk(item.WFNFileName, "agsfnt" + (item.ID - 1) + ".wfn");
-                            }
-                            if (File.Exists(item.TTFFileName))
-                            {
-                                _agsEditor.RenameFileOnDisk(item.TTFFileName, "agsfnt" + (item.ID - 1) + ".ttf");
-                            }
-                            item.ID--;
-                        }
-                    }
-                    if (_documents.ContainsKey(_itemRightClicked))
-                    {
-                        _guiController.RemovePaneIfExists(_documents[_itemRightClicked]);
-                        _documents.Remove(_itemRightClicked);
-                    }
-                    _agsEditor.CurrentGame.Fonts.Remove(_itemRightClicked);
-                    _agsEditor.CurrentGame.FilesAddedOrRemoved = true;
-                    Factory.NativeProxy.OnFontDeleted(_agsEditor.CurrentGame, removingID);
-                    Factory.NativeProxy.GameSettingsChanged(_agsEditor.CurrentGame);
-                    RePopulateTreeView();
-                    FontTypeConverter.SetFontList(_agsEditor.CurrentGame.Fonts);
+                    DeleteFont(_agsEditor.CurrentGame, _itemRightClicked as AGS.Types.Font);
                 }
             }
             else
             {
-                if (controlID != TOP_LEVEL_COMMAND_ID)
+                if (controlID.StartsWith(FONT_FILE_ITEM_PREFIX))
+                {
+                    string fontFileName = controlID.Substring(FONT_FILE_ITEM_PREFIX.Length);
+                    FontFile chosenFontFile = FindFontFileByName(fontFileName);
+                    ShowOrAddPane(chosenFontFile);
+                }
+                else if (controlID.StartsWith(FONT_ITEM_PREFIX))
+                {
+                    AGS.Types.Font chosenFont = _agsEditor.CurrentGame.Fonts[Convert.ToInt32(controlID.Substring(FONT_ITEM_PREFIX.Length))];
+                    ShowOrAddPane(chosenFont);
+                }
+                else if (controlID.StartsWith(FONT_ITEM_REF_PREFIX))
                 {
-                    AGS.Types.Font chosenFont = _agsEditor.CurrentGame.Fonts[Convert.ToInt32(controlID.Substring(3))];
+                    AGS.Types.Font chosenFont = _agsEditor.CurrentGame.Fonts[Convert.ToInt32(controlID.Substring(FONT_ITEM_REF_PREFIX.Length))];
                     ShowOrAddPane(chosenFont);
                 }
             }
         }
 
+        private FontFile FindFontFileByName(string fontFileName)
+        {
+            return _agsEditor.CurrentGame.FontFiles.Where(ff => ff.FileName == fontFileName).FirstOrDefault();
+        }
+
+        private void ShowOrAddPane(FontFile chosenFontFile)
+        {
+

```

Comment:
If this function is meant to be empty but needs to exist, I suggest adding a comment here, something like "panes only exist for Font for now, but we may call this method for a node that may be a FontFile, just do nothing".